### DISCUSSION TITLE
Stay on the selected ruleset (gamemode) when opening a beatmap from notifications or beatmap listing

### DISCUSSION THREAD

**Discussion by Nekomosh004** [ppy/osu#20445] [Ideas]
After I download a beatmap that is for osu! standard while I'm on a different ruleset other than standard (example, I'm on catch), it switches the ruleset to osu! standard when I open the beatmap using the notification message. I kind of understand why this is a good idea, because in the first place, the downloaded beatmap is for osu! standard and not for catch.
I tested downloading a catch specific beatmap while osu! standard ruleset is selected and it switched it to catch when I open the notification. I guess this also is an expected behavior when I download a taiko or mania beatmap while osu! standard ruleset is selected.

https://user-images.githubusercontent.com/82471957/192079507-0c148420-8a4c-400b-aba4-c06431672be2.mp4

Edit: It also happens when I open a standard map from the beatmap listing

https://user-images.githubusercontent.com/82471957/192080049-89f36a4f-5eca-4e57-90e1-633fd6ab0098.mp4

Just a suggestion: If the "Show converted" option is ticked, downloading a osu! standard beatmap and opening it from the notification section should not switch the ruleset to osu! standard.

![image](https://user-images.githubusercontent.com/82471957/192079581-c483826b-d980-4338-8989-7ab1c295d705.png)

Its not that a hassle to switch between gamemodes, its just that I feel its better to be on the same gamemode because I have converted beatmaps enabled. If its disabled, then I think its expected to switch rulesets because it won't be showing up.

Thank you for considering c:

**Comment by peppy** [comment]
I can agree with this one. It relates to the flow I was [trying to fix](https://github.com/ppy/osu/issues/20528#issuecomment-1265306626) recently, and would be fixed by the proposal I made (when coming from a beatmap were a ruleset is not defined, ie. the cases you mention, it should maintain the local ruleset if possible).

It's not a simple fix so will likely come later with a larger refactor, but is definitely the correct direction to take.
